Manual: Fix references to issues (#2135)
well-typed/hs-bindgen: manual/low-level/translation/generated-names.md

_anywhere_ in a Haskell name). By default we escape such characters.

Note that in particular any character which does not satisfy
[`isAlphaNum`][hackage:base:isAlphaNum] is always invalid.

> [!NOTE]
> We could instead _drop_ such invalid characters, but since this could result

> > in name clashes we do not do this by default.

Reviewer: ```suggestion
> in name clashes we avoid dropping characters by default.
```

(As a side note, if there were name clashes, we would detect them with out shiny new name mangler clash-detection traversal; I don't think this is relevant here, just saying).

[`isAlphaNum`][hackage:base:isAlphaNum] is always invalid.

> [!NOTE]
> We could instead _drop_ such invalid characters, but since this could result
> in name clashes we do not do this by default.

> [!NOTE]
> For this _particular_ example we could generate a better name if we
> transformed it to [NFC][unicode:nfc] first. Indeed, `gcc` (but not `clang`)
> will issue a warning that this name is not in NFC.

> > <https://github.com/well-typed/hs-bindgen/issues/560>

Author: Since [#560] is closed as completed, I thought generated names would now always be in NFC. However, looking at the related golden test case, we seem to still generate identifiers `Adio'0301s` etc., so are we not normalising after all? I'll double check and potentially revert part of my change.

Reviewer: Yes, I was wondering about this change, and checked the issue. But it was closed, so I assumed it is "fixed".

Author: Unless I'm missing something we're not doing Unicode Normalization. I now re-added that note to the manual and will leave a brief clarifying comment on the closed issue as well to avoid future confusion.
